Objects/unicodeobject.c

```diff
@@ -3922,10 +3922,6 @@ PyUnicode_FSDecoder(PyObject* arg, void* addr)
     }
 
     if (PyUnicode_Check(path)) {
-        if (PyUnicode_READY(path) == -1) {
```

Comment:
It's not an error but not a must. After the `if ... else ...` we get a code path to ready it.